osf/models/mixins.py

```diff
@@ -2074,7 +2074,7 @@ def _get_spam_content(self, saved_fields):
         return b' '.join(content).decode()
 
     def check_spam(self, user, saved_fields, request_headers):
-        if not settings.SPAM_CHECK_ENABLED:
+        if not user:  # in case of tests and staff admin operations
```

Comment:
Note, line 2097 on are now pointless, since `do_check_spam` no longer returns a meaningfully truth-y value in most cases. We need to determine the best way to re-locate that logic.

Reply:
I moved it into the overall spam services checking task now it checks if a resource has a `suspend_spam_user` or `_check_spam_user` method and then suspends the spam content author.